### DISCUSSION TITLE
giving the admins control to choose which admin menu items should be shown in the user bar menu

### DISCUSSION THREAD

**Discussion by hazho** [wagtail/wagtail#9302] [Ideas]
currently the capability of adding/registering a menu item to user bar is only by the developers, what would be great though is to give sort of site-setting per user as a front-end preferences (i.e. storing the user layout in the local storage or something similar)..!

**Comment by lb-** [comment]
@hazho I think this would be possible with a customisation, you could build a site setting that allows this kind of thing to be customised and then use the existing hook to read in that data.

However... need to check you can get the site from `request` only - otherwise you will need https://github.com/wagtail/wagtail/issues/3847

I think this level of granular admin control would be best served as a customisation and then maybe a package, but that's just my opinion.

Another way to think about this is that it is mostly permissions based, and hence the admin kind of already has control over this by what users have what permissions. To an extent.

**Reply by hazho** [reply]
We already did that, but the problem is that this approach is a (site level setting), and I think that User preferences should be stored in the browser's storage, not in the DB, unless we track every actions of each users, in that case we could simply add a field called (preferences) to the user activity model, and get and set each preference in the cookies or storage in the browser.
specifically for the permissions you mentioned, this feature should be part of built-ins admin app.
even pagination-size value and many other settings should be moved to front-end and per-user setting

**Reply by lb-** [reply]
Ok. That's a bit different to what I understood.
